### ISSUE TITLE
support  [r| { a,b :: T } |] syntax

### ISSUE THREAD

**Issue by aavogt** [nikita-volkov/record#19]
Currently it fails with `Parser failure: Failed reading: satisfy. Contexts: ["type'","'('"].` which could be a better message too.

**Comment by rabipelais** [comment]
Would this be equivalent to `[r| { a :: T, b :: T } |]`?

**Comment by aavogt** [comment]
yes just like how in haskell you can write `data T t = Con { a, b :: t }`

**Comment by nikita-volkov** [comment]
Concerning the error message see https://github.com/nikita-volkov/record/issues/4.

Concerning getting this syntax supported, I'm conflicted about this. On one hand, as long as it concerns only the type-level syntax, I don't see this creating any ambiguity. OTOH I don't see much usefulness in this syntax and instead see it as more of pertaining to feature bloat. So my vote is negative on this one. I won't stand in your way if you decide to implement it, however I must inform you that the preprocessor reimplementation of the library will initially be lacking this feature either way, since it's a complete rewrite of the related part of the library.
